### DISCUSSION TITLE
Navigation Tree not filtering correctly items when current usergroup is based on permissions  by tags

### DISCUSSION THREAD

**Discussion by KingRial** [requarks/wiki#3548] [Error / Bug Report]
**Describe the bug**
The navigation panel seems unable to filter correctly tree items when current usergroup is using permissions by tags

**To Reproduce**
1. Go into administration and create a custom usergroup
2. Configure a single "Page rule" based on tag "public" (See screenshot)
3. Create page A with the tag "public"
4. Create page B without any tag
5. Access with a user based on custom usergroup
6. Access frontend and on the navigation (Tree site) you don't see any page

**Expected behavior**
You should see the page A in the navigation and not the page B

**Screenshots**
Example of usegroup configuration
![image](https://user-images.githubusercontent.com/886518/92730870-a8aeb800-f374-11ea-8402-3815a7ec31c8.png)

**Host Info (please complete the following information):**
 - OS: Docker
 - Wiki.js version: 2.5.126
 - Database engine: postgres 9.7

**Additional context**
I tried to play a little with the code and created a workaround (https://github.com/Clevermind-Think-Mint/wiki/commit/11fcb24a4284a7d5980332b8e06119349e6569c2)

The concept is:
- during the job "tree-rebuild", tags are also added to each entry
- tags are then used to check access to the resource when showing the navigation tree items

It's far from being decent and it lacks the migration to update correctly the DB:
```sql
-- Navigation will be filtered by tags permissions
ALTER TABLE "pageTree" ADD COLUMN tags json;
```

However it might be useful to find a better solution.
Any opinion is welcome before considering to make a pull request!

**Comment by dominicgermain** [comment]
Same issue here... Trying to build a multi departement Wiki with branches of the tree for each of them and using "tags" to provide special access to pages that belongs to others...  So there is two things that are problematic with pages visible through tags :   

1. not visible in the tree
1. not visible in search results neither

The only way is to provide the path directly to other users... not really an option over the time.

**Comment by yusufgungor** [comment]
Same issue here +1

Is there any plan for this?

**Comment by rooso** [comment]
Same issue here +1
Would like to allow some customers to access specific pages.

**Comment by marcela-fernanda** [comment]
Same issue here +1

**Comment by smenzer** [comment]
+1 same issue

**Comment by machinaeXphilip** [comment]
could this be related to the issue i describe here: https://github.com/requarks/wiki/discussions/6026 ?

**Comment by kamph** [comment]
+1 same issue

**Comment by AliciaBytes** [comment]
This is currently a blocker for me too +1
